frontend/src/lib/components/PropertyFilters/PropertyFilters.js

```diff
@@ -33,7 +33,8 @@ function FilterRow({ item, index, filters, logic, pageKey }) {
                 {key ? (
                     <Button type="primary" shape="round" style={{ maxWidth: '85%' }}>
                         <span style={{ width: '100%', overflow: 'hidden', textOverflow: 'ellipsis' }}>
-                            {keyMapping[key]?.label || key} {operatorMap[operator || 'exact'].split(' ')[0]} {value}
+                            {keyMapping[type === 'element' ? 'element' : 'event'][key]?.label || key}{' '}
+                            {operatorMap[operator || 'exact'].split(' ')[0]} {value}
```

Comment:
I think with the `keyMapping[type === 'element' ? 'element' : 'event'][key]?.label` code being used here and there, it might be worth extracting this into some `getNiceTitleForProperty(type, key)` function. 

I wouldn't block this PR for that though.